src/librustc/lint/context.rs

```diff
@@ -52,12 +52,16 @@ pub struct LintStore {
     /// added by a plugin.
     lints: Vec<&'static Lint>,
 
-    /// Trait objects for each lint pass.
-    /// This is only `None` while performing a lint pass.
-    pre_expansion_passes: Option<Vec<EarlyLintPassObject>>,
-    early_passes: Option<Vec<EarlyLintPassObject>>,
-    late_passes: Lock<Option<Vec<LateLintPassObject>>>,
-    late_module_passes: Vec<LateLintPassObject>,
+    /// Constructor functions for each variety of lint pass.
+    ///
+    /// These should only be called once, but since we want to avoid locks or
```

Comment:
Module passes will be created once per module, so this comment isn't true for them.

Reply:
Ah, yeah, I'll leave a comment to that effect. We don't actually do that today anyway I believe -- at least, the previous code never had an impl of `fresh_late_pass`, it always panic'd...

Reply:
All the late module passes are hardcoded in the compiler so they don't appear in this list, hence they don't need a real impl of `fresh_late_pass`, but I guess we shouldn't allow plugins to add to `late_module_passes` either since that might not won't work for incremental compilation.